system/Database/BaseBuilder.php

```diff
@@ -2698,7 +2698,7 @@ protected function objectToArray($object)
 
         foreach (get_object_vars($object) as $key => $val) {
             // There are some built in keys we need to ignore for this conversion
-            if ((! is_object($val) && ! is_array($val) && $key !== '_parent_name') || is_a($val, RawSql::class)) {
+            if (((! is_object($val) || $val instanceof RawSql) && ! is_array($val) && $key !== '_parent_name')) {
```

Comment:
What's the difference?
Can you add a test case?

Reply:
I was just keeping intact the logic that was there already. They had it before that $key !== '_parent_name' under any circumstances. I'm really not sure what its there for. 

The original logic was: if (! is_object($val) && ! is_array($val) && $key !== '_parent_name')

I added: (is not and object unless its RawSql)

Reply:
I was just making sure we ignored:

```php
// There are some built in keys we need to ignore for this conversion
```

Apparently some objects must have a property _parent_name that needs to be ignored.

Reply:
Do you mean?

```php
        foreach (get_object_vars($object) as $key => $val) {
            // There are some built in keys we need to ignore for this conversion
            if (! is_object($val) && ! is_array($val) && $key !== '_parent_name') {
                if ((! $val instanceof RawSql)) {
                    $array[$key] = $val;
                }
            }
        }
```

Reply:
I got it.
Can you add the following test case?
```diff
--- a/tests/system/Database/Builder/InsertTest.php
+++ b/tests/system/Database/Builder/InsertTest.php
@@ -13,6 +13,7 @@ namespace CodeIgniter\Database\Builder;
 
 use CodeIgniter\Database\Exceptions\DatabaseException;
 use CodeIgniter\Database\Query;
+use CodeIgniter\Database\RawSql;
 use CodeIgniter\Test\CIUnitTestCase;
 use CodeIgniter\Test\Mock\MockConnection;
 
@@ -85,6 +86,21 @@ final class InsertTest extends CIUnitTestCase
         $this->assertSame($expectedBinds, $builder->getBinds());
     }
 
+    public function testInsertObjectWithRawSql()
+    {
+        $builder = $this->db->table('jobs');
+
+        $insertData = (object) [
+            'id'   => 1,
+            'name' => new RawSql('CONCAT("id", \'Grocery Sales\')'),
+        ];
+        $builder->testMode()->insert($insertData, true);
+
+        $expectedSQL = 'INSERT INTO "jobs" ("id", "name") VALUES (1, CONCAT("id", \'Grocery Sales\'))';
+
+        $this->assertSame($expectedSQL, str_replace("\n", ' ', $builder->getCompiledInsert()));
+    }
+
     /**
      * @see https://github.com/codeigniter4/CodeIgniter4/issues/5365
      */
```

Reply:
Done